target/oneNoteApi.js

```diff
@@ -291,6 +291,31 @@ var OneNoteApi = (function (_super) {
         return this.requestPromise(url, form.asBlob(), form.getContentType());
     };
     /**
+     * BatchRequests
+     **/
+    OneNoteApi.prototype.batchRequests = function (batchRequests) {
+        this.enableBetaApi();
+        var boundaryName = "batch_" + Math.floor(Math.random() * 1000);
+        var contentType = "Content-Type: application/http";
+        var contentTransferEncoding = "Content-Transfer-Encoding: binary";
+        var data = "";
+        batchRequests.forEach(function (batchRequest) {
+            var req = "";
+            req += "--" + boundaryName + "\r\n";
+            req += "Content-Type: application/http" + "\r\n";
+            req += "Content-Transfer-Encoding: binary" + "\r\n";
+            req += "\r\n";
+            req += batchRequest.httpMethod + " " + batchRequest.uri + " " + "HTTP/1.1" + "\r\n";
+            req += "Content-Type: " + batchRequest.contentType + "\r\n";
+            req += "\r\n";
+            req += batchRequest.content + "\r\n";
+            req += "\r\n";
+            data += req;
+        });
+        data += "--" + boundaryName + "--\r\n";
+        return this.requestBasePromise("/$batch", data, 'multipart/mixed; boundary="' + boundaryName + '"', "POST").then(this.disableBetaApi.bind(this));
```

Comment:
Race condition: Possible for another interleaving API call to hit beta api. Consider implementing enableBetaApiForNextRequest()